### ISSUE TITLE
Iterative Upscale (latent) error with self attention guidance

### ISSUE THREAD

**Issue by jatgoodwin** [ltdrdata/ComfyUI-Impact-Pack#372]
It seems if the latent ratio is too square (512x520 or close) I get an error but only if self attention guidance is applied. 
![Screenshot 2023-12-15 at 10-23-01 ComfyUI](https://github.com/ltdrdata/ComfyUI-Impact-Pack/assets/8681973/55887ab0-31a9-4de6-90ec-602cccbe8623)
If I set the empty latent image too something like 512x768 it works just fine.

```IterativeLatentUpscale[1/3]: 606.7x597.3 (scale:1.17)
  0%|                                                                                           | 0/30 [00:00<?, ?it/s]
ERROR:root:!!! Exception during processing !!!
ERROR:root:Traceback (most recent call last):
  File "C:\ai\ComfyUI_windows_portable\ComfyUI\execution.py", line 153, in recursive_execute
    output_data, output_ui = get_output_data(obj, input_data_all)
  File "C:\ai\ComfyUI_windows_portable\ComfyUI\execution.py", line 83, in get_output_data
    return_values = map_node_over_list(obj, input_data_all, obj.FUNCTION, allow_interrupt=True)
  File "C:\ai\ComfyUI_windows_portable\ComfyUI\execution.py", line 76, in map_node_over_list
    results.append(getattr(obj, func)(**slice_dict(input_data_all, i)))
  File "C:\ai\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-Impact-Pack\modules\impact\impact_pack.py", line 1040, in doit
    current_latent = upscaler.upscale_shape(step_info, current_latent, new_w, new_h, temp_prefix)
  File "C:\ai\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-Impact-Pack\modules\impact\core.py", line 1664, in upscale_shape
    refined_latent = nodes.KSampler().sample(model, seed, steps, cfg, sampler_name, scheduler,
  File "C:\ai\ComfyUI_windows_portable\ComfyUI\nodes.py", line 1299, in sample
    return common_ksampler(model, seed, steps, cfg, sampler_name, scheduler, positive, negative, latent_image, denoise=denoise)
  File "C:\ai\ComfyUI_windows_portable\ComfyUI\nodes.py", line 1269, in common_ksampler
    samples = comfy.sample.sample(model, noise, steps, cfg, sampler_name, scheduler, positive, negative, latent_image,
  File "C:\ai\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-Impact-Pack\modules\impact\sample_error_enhancer.py", line 22, in informative_sample
    raise e
  File "C:\ai\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-Impact-Pack\modules\impact\sample_error_enhancer.py", line 9, in informative_sample
    return original_sample(*args, **kwargs)  # This code helps interpret error messages that occur within exceptions but does not have any impact on other operations.
  File "C:\ai\ComfyUI_windows_portable\ComfyUI\comfy\sample.py", line 100, in sample
    samples = sampler.sample(noise, positive_copy, negative_copy, cfg=cfg, latent_image=latent_image, start_step=start_step, last_step=last_step, force_full_denoise=force_full_denoise, denoise_mask=noise_mask, sigmas=sigmas, callback=callback, disable_pbar=disable_pbar, seed=seed)
  File "C:\ai\ComfyUI_windows_portable\ComfyUI\comfy\samplers.py", line 714, in sample
    return sample(self.model, noise, positive, negative, cfg, self.device, sampler, sigmas, self.model_options, latent_image=latent_image, denoise_mask=denoise_mask, callback=callback, disable_pbar=disable_pbar, seed=seed)
  File "C:\ai\ComfyUI_windows_portable\ComfyUI\comfy\samplers.py", line 620, in sample
    samples = sampler.sample(model_wrap, sigmas, extra_args, callback, noise, latent_image, denoise_mask, disable_pbar)
  File "C:\ai\ComfyUI_windows_portable\ComfyUI\comfy\samplers.py", line 559, in sample
    samples = self.sampler_function(model_k, noise, sigmas, extra_args=extra_args, callback=k_callback, disable=disable_pbar, **self.extra_options)
  File "C:\ai\ComfyUI_windows_portable\python_embeded\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "C:\ai\ComfyUI_windows_portable\ComfyUI\comfy\k_diffusion\sampling.py", line 695, in sample_dpmpp_3m_sde_gpu
    return sample_dpmpp_3m_sde(model, x, sigmas, extra_args=extra_args, callback=callback, disable=disable, eta=eta, s_noise=s_noise, noise_sampler=noise_sampler)
  File "C:\ai\ComfyUI_windows_portable\python_embeded\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "C:\ai\ComfyUI_windows_portable\ComfyUI\comfy\k_diffusion\sampling.py", line 655, in sample_dpmpp_3m_sde
    denoised = model(x, sigmas[i] * s_in, **extra_args)
  File "C:\ai\ComfyUI_windows_portable\python_embeded\lib\site-packages\torch\nn\modules\module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "C:\ai\ComfyUI_windows_portable\ComfyUI\comfy\samplers.py", line 283, in forward
    out = self.inner_model(x, sigma, cond=cond, uncond=uncond, cond_scale=cond_scale, model_options=model_options, seed=seed)
  File "C:\ai\ComfyUI_windows_portable\python_embeded\lib\site-packages\torch\nn\modules\module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "C:\ai\ComfyUI_windows_portable\ComfyUI\comfy\samplers.py", line 273, in forward
    return self.apply_model(*args, **kwargs)
  File "C:\ai\ComfyUI_windows_portable\ComfyUI\comfy\samplers.py", line 270, in apply_model
    out = sampling_function(self.inner_model, x, timestep, uncond, cond, cond_scale, model_options=model_options, seed=seed)
  File "C:\ai\ComfyUI_windows_portable\ComfyUI\comfy\samplers.py", line 261, in sampling_function
    cfg_result = fn(args)
  File "C:\ai\ComfyUI_windows_portable\ComfyUI\comfy_extras\nodes_sag.py", line 152, in post_cfg_function
    degraded = create_blur_map(uncond_pred, uncond_attn, sag_sigma, sag_threshold)
  File "C:\ai\ComfyUI_windows_portable\ComfyUI\comfy_extras\nodes_sag.py", line 68, in create_blur_map
    mask.reshape(b, *mid_shape)
RuntimeError: shape '[1, 11, 11]' is invalid for input of size 100```

**Comment by ltdrdata** [comment]
This is a issue of the SAG.

If the latent dimension is `[65, 65]`(pixel 520x520), creating the attention dimension as `[1, 8, 81, 81]` makes reshaping impossible. It seems like there is a need for a fix in the `create_blur_map` function.

(https://github.com/comfyanonymous/ComfyUI/blob/adc40e3d7bc612e81874cf9f5738bda0e17ce0a3/comfy_extras/nodes_sag.py#L56)

**Comment by ltdrdata** [comment]
This is fixed on [172984db0175845c1a16bc3100fed0e46b42f604](https://github.com/comfyanonymous/ComfyUI/commit/172984db0175845c1a16bc3100fed0e46b42f604)
